app/src/main/java/com/infomaniak/drive/utils/PreviewUtils.kt

```diff
@@ -161,3 +168,39 @@ private fun Context.printPdf(
         file != null -> printFile()
     }
 }
+
+fun downloadFile(
+    externalOutputFile: IOFile,
+    fileModel: File,
+    shouldBePdf: Boolean,
+    onProgress: (progress: Int) -> Unit,
+) {
+    if (externalOutputFile.exists()) externalOutputFile.delete()
+
+    val downloadUrl = fileModel.downloadUrl() + if (fileModel.isOnlyOfficePreview()) "?as=pdf" else ""
+    val downloadProgressInterceptor = DownloadOfflineFileManager.downloadProgressInterceptor(onProgress = onProgress)
+
+    DownloadOfflineFileManager.downloadFileResponse(downloadUrl, downloadInterceptor = downloadProgressInterceptor).use {
+        if (!it.isSuccessful) {
+            val errorCode = JsonParser.parseString(it.body?.string()).asJsonObject.getAsJsonPrimitive("error").asString
+            if (errorCode == "password_protected_error") {
+                throw PasswordProtectedException()
+            } else {
+                throw Exception("Download error")
+            }
+        }
+
+        if (shouldBePdf && it.body?.contentType()?.toString() != "application/pdf") {
+            throw UnsupportedOperationException("File not supported")
+        }
+
+        createTempFile(it, externalOutputFile)
+        onProgress(PROGRESS_COMPLETE)
+    }
+}
+
+private fun createTempFile(response: Response, file: IOFile) {
```

Comment:
If `createTempFile` is only used in `downloadFile`, maybe we can create an inner function in the `downloadFile` method ?

Reply:
I found the downloadFile() function already a bit consequent, so I'd rather not encumber it more